### PR TITLE
Forecasting cryptocurrency returns with Twitter sentiment

### PR DESCRIPTION
**Pull request by HuifangYeo**

### CODE REVIEW

File: fun/twitter/util/utils.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        # we generate index from the last date for a period equivalent to the size of the forecast
+        last_date = train_data.tail(1).index.get_level_values("date").to_pydatetime()[0]
+        date_indices = pd.date_range(
+            start=last_date, periods=(forecast_steps + 1), closed="right"
+        )
+        pred_df = pd.DataFrame(
+            pred, index=date_indices, columns=train_data.columns,
+        ).reset_index()
+
+#         accuracy_df = pd.DataFrame()
```
Comment (fabiencelier) [inline fun/twitter/util/utils.py:165]: You should not keep commented code

**Comment by fabiencelier** [comment]
> We can do so by query the atoti datacube.

Don't use the name "datacube", this is a word used when doing distribution in atoti which you are not doing

**Review by fabiencelier — COMMENTED** [review]
Just doing a review of the main notebook with atoti stuff. I have not read every thing in details, just going quickly through it

**Comment by fabiencelier** [comment]
>     types={
>        "Tweet volume": tt.types.FLOAT,
>        "Retweet total": tt.types.FLOAT,
>        "Retweet average": tt.types.FLOAT,
>        "Followers total": tt.types.FLOAT,
>        "Followers average": tt.types.FLOAT,
>        "Favorite total": tt.types.FLOAT,
>        "Favorite average": tt.types.FLOAT,
>        "Polarity total": tt.types.FLOAT,
>        "Polarity average": tt.types.FLOAT,
>        "Negative": tt.types.FLOAT,
>        "Neutral": tt.types.FLOAT,
>        "Positive": tt.types.FLOAT,
>        "Bullish ratio": tt.types.FLOAT,
>        "Subset": tt.types.STRING,
>    },
    
Why specify the types ? can't you just used double and the string should be autodetected ?

**Comment by fabiencelier** [comment]
> C:\Apps\miniconda3\envs\atoti\lib\site-packages\statsmodels\tsa\stattools.py:662: FutureWarning: fft=True will become the default after the release of the 0.12 release of statsmodels. To suppress this warning, explicitly set fft=False.
  warnings.warn(
  
 You have some deprecation warning you should ged rid off

**Comment by fabiencelier** [comment]
> We are unable to interact with the data in vector format.

This is not true, there are many atoti functions for vectors !! This comment gives the impression that atoti is limited with vectors while it is very good (IMHO)

**Comment by fabiencelier** [comment]
> Note that we won't be plotting a lot of charts along this notebook for our analysis as what most analysts would normally do.

This also looks like an atoti limitation

**Comment by fabiencelier** [comment]
nan

You have some Nan values in your query result

**Comment by fabiencelier** [comment]
chart

why not use quick filter for changing the filters ?

**Comment by fabiencelier** [comment]
> ```python
>m["Trend Sign"] = tt.where((m["Returns"] - m["Prev day returns"]) < 0, "📉", tt.where((m["Returns"] - m["Prev day returns"]) > 0, "📈", "➡") )
>```

That symbol measure :laughing:

**Comment by fabiencelier** [comment]
> Accessing the vectors by using the index of the parameter hierarchy - Lag Index. \ Notice here we have created the level of significant as a measure alpha? We will be using this measure to perfrom what-if simulation later.

You sometime have some random `\` in you notebook text

**Comment by HuifangYeo** [comment]
> > We can do so by query the atoti datacube.
> 
> Don't use the name "datacube", this is a word used when doing distribution in active pivot (and soon in atoti) which you are not doing

What would be the correct name then? Do we just say cube?

**Comment by HuifangYeo** [comment]
> > ```
> > types={
> >    "Tweet volume": tt.types.FLOAT,
> >    "Retweet total": tt.types.FLOAT,
> >    "Retweet average": tt.types.FLOAT,
> >    "Followers total": tt.types.FLOAT,
> >    "Followers average": tt.types.FLOAT,
> >    "Favorite total": tt.types.FLOAT,
> >    "Favorite average": tt.types.FLOAT,
> >    "Polarity total": tt.types.FLOAT,
> >    "Polarity average": tt.types.FLOAT,
> >    "Negative": tt.types.FLOAT,
> >    "Neutral": tt.types.FLOAT,
> >    "Positive": tt.types.FLOAT,
> >    "Bullish ratio": tt.types.FLOAT,
> >    "Subset": tt.types.STRING,
> > ```
> > 
> > 
> > },
> 
> Why specify the types ? can't you just used double and the string should be autodetected ?

Hi @fabiencelier,

without quoting the types, I will run into unexpected conversion error during data loading into scenarios.

**Comment by HuifangYeo** [comment]
> nan
> 
> You have some Nan values in your query result

@fabiencelier, these columns are being used as a place holder. Alternative, can I add extra columns to datastore after it has been created? I didn't want to create too many stores.

**Comment by APolyakova** [comment]
If PR84 is merged - pls consider using the new location: under "notebooks" folder.

**Review by arielibaba — APPROVED** [review]
Agreed on modifications made by Hui Fang.

**Pull request merged**